provider/utils.go

```diff
@@ -15,6 +15,22 @@ func contains(arr []string, val string) bool {
 	return false
 }
 
+func Flatten(m map[string]interface{}) map[string]interface{} {
```

Comment:
and how will this work when we look at actual logs? I am thing that log will not be flattened or do you intend to do something similar there as well?

Reply:
yes I'll flatten the log

Reply:
as discussed, we're doing this now in the javascript side